### ISSUE TITLE
Ross 128b "free machines" solution

### ISSUE THREAD

**Issue by richardhendricks** [GTNewHorizons/GT-New-Horizons-Modpack#4958]
I've been thinking about #4916 and have a proposal for a solution.

tl;dr
Instead of placing actual GT machines on the surface, place a new block that uses the machine's texture, but when broken does not always return the machine. 

Details
This block would be like the Hives from forestry - you cannot look inside with holoinventory since the "contents" are only generated when it is broken. It cannot be picked up  by itself.

Blocks UUID is only NBT data for the block, sets the block "type" according to the config options below.  SO there could be multiple "ruined pulverizers" but each UUID could give different tiers of items.

Texture could just be the base texture. (will need to figure out how to deal with texturepacks etc)

When broken, the block goes over the loot tables and drops either the "real" block or a couple of items from the "parts" loot table.

With a generic configuration option, this system could be useful for a lot of other things. Like finding astronaut bodies on planets etc that can give various items/parts as well. Could recycle things like skulls into giving a reward too.

Must be placeable in creative mode, with a way of setting the UUID.

Maybe add an option for "explode chance if broken" and "explosion force".

Configuration options - These make the system generic so it can be used for other things.
Block's name - will need support for languages
Block's UUID? (specific to this mod)
Block's model mesh
Block's texture
Explode chance
Explosion force
Block has two-branch loot table system
  - Top level has weights for each branch to drop (say 5 for first, 95 for second)
    - One branch can drop the actual machine (5% of the time)
    - One branch can drop parts of the machine (95%), including input items and output items, with percentage chances for each item (setup so that when an item is selected, it is removed from the loot table for future passes) and a limited number of passes.

Sample config
Name: Ruined Pulverizer
UUID: (custom UUID from an online uuid generator)
Mesh: Simple block
Texture: Pulverizer
Explode chance: 10 ( 1 in 10 chance)
Explosion force: 5
Loot Table:
 weight 5, # of rolls 1
  - HV pulverizer
 weight 95, # of rolls 1-3
  - 50 HV machine hull
  - 25 HV circuit
  - 20 HV circuit
  - 15 HV motor
  - 10 industrial diamond
  - 10 (ore of some kinds)
  - 10 (crushed ore of some kinds)

**Comment by bartimaeusnek** [comment]
as i have stated in the past. i wont change this. if you want to do that, go ahead the code is MIT. but either way of "fixing" this wont be beneficial; since right now, the machines are in my opinion perfectly tiered, it takes less time to actually build the machine you want, if you use drilling rigs and iv machinery OR MV/HV machinery and an AE2 system, as it takes you to gather the machines from the planet. nerfing this mechanic or making it even more complicated is just a useless timesink that states no benefit other than listening to a few people that do not represent the major part of the community. Especially the beta testers that actually tested my stuff over a longer period of time told me that the droprate of actual good machinery compared to the invested time is fun, but not worth it, timewise.

**Comment by AlexBucket** [comment]
A hut appearing with a machine in it might be 1:3072, but what’s still 100% more chance than finding one randomly on any other planet. It’s far from “perfectly tiered”.

**Comment by AlexBucket** [comment]
Adding a chance of a free machine doesn’t make the planet any less boring, just more broken.

**Comment by boubou19** [comment]
It is given to the player the choice of either pushing his luck to find 3 IV machines and a Nq generator, or making the player make a better use of the time he could have spent searching IV machines by making more infrastructure. By the time you reach T3 planets you are already at IV, meaning you could make more than 3 machines per hour. I guess the """issue""" here is the fact that you can have 1 Nq gen per hour: it can seem a bit high but hey it's only 1 IV amp. There is plenty of ways to make amps of IV at IV. And we should keep in mind that flying to explore Ross128b is generating a lot of chunks so it is not viable on a server because of tps drops it would cause, and in solo, well if the player wants he still can spawn the items if he doesn't want to craft so no biggie there too.

**Comment by observeroftime02** [comment]
I just want to add, I've spent a fair amount (several hours) on Ross exploring the ruins, and I've yet to run into a Nq gen or a lightning rod...

**Comment by boubou19** [comment]
if a player drops tps intentionnally  he should be punished because by that he lowers other players' pack experience. I don't see in what Indium veins on Ross 128b, the rarest veins of the planet, is better than the classical way of getting indium? Because you get maybe half a stack of a vein on Ross128b so you spend a lot of time to get some Indium meanwhile the classical process could produce the same amount. As stated above, the fact that a nq gen need zpm circuits shouldn't interfer because there is ways to get IV amps before Ross128b which don't need zpm circuits. It's as if players complain about the fact that a zero point module is broken, despite the fact it's drop rate is very rare (+ you can get one for free somewhere in the quests).
